### PR TITLE
Larq ref impl

### PR DESCRIPTION
**Pull request by andrewstanfordjason**

### CODE REVIEW

**Comment by andrewstanfordjason** [comment]
please review:
lib_nn/api/OutputTransformFn.hpp 
lib_nn/src/asm/output_transform_fn_binary.S 
lib_nn/src/asm/output_transform_fn_int8_clamped.S
lib_nn/src/cpp/OutputTransformFn.cpp
test/shared/include/RefOps.hpp
test/shared/src/ops/ref/larq.cpp 
test/unit_test/src/test_Conv2dRegression.cpp

The rest of the files are mostly benign or formatting/comments

File: lib_nn/src/cpp/OutputTransformFn.cpp [hunk 1]
```diff
@@ -522,8 +504,130 @@ int8_t *OT_int8::output_transform_fn(int8_t *Y, VPURingBuffer *A,
...
+      (int32_t)VPU_INT16_EPV);
+
+  // Thge 3 is due to the serialisation of 3 arrays in chunks of VPU_INT16_EPV
+  // elements
+  int16_t *cur_post_activation_offset =
+      offsets_multipliers_and_biases + output_channel_group * VPU_INT16_EPV * 3;
+
+  int16_t *cur_post_activation_mul = cur_post_activation_offset + output_count;
+
+  int16_t *cur_post_activation_bias = cur_post_activation_mul + output_count;
```
Comment (ghost) [inline lib_nn/src/cpp/OutputTransformFn.cpp]: Are we using these values?

Comment (andrewstanfordjason) [reply lib_nn/src/cpp/OutputTransformFn.cpp]: removed

**Review by ghost — APPROVED** [review]
+1

**Review by henkmuller — APPROVED** [review]
Looks good.

I am not sure why multipliers and biases have been pulled apart - I thought it was more efficient when they were interleaved in the assembly kernel.

For future reference - I would prefer it if we can run the auto-layouter between reviews, so we never have to review spacing changes.

**Comment by andrewstanfordjason** [comment]
the multipliers and biases are interleaved at the asm level still. I just extracted the function that does it to separate the responsibilities into separate functions.

And yes, formatting shouldn't cause a mess! the last few PRs were not formatted correctly at the time of merge so the formatting has drifted.

**Pull request merged**